crates/rpc/rpc/src/debug.rs

```diff
@@ -613,8 +643,20 @@ where
 #[async_trait]
 impl<Provider, Eth> DebugApiServer for DebugApi<Provider, Eth>
 where
-    Provider: BlockReaderIdExt + HeaderProvider + ChainSpecProvider + 'static,
-    Eth: EthApiSpec + SpawnBlocking + LoadState + LoadPendingBlock + 'static,
+    Provider: BlockReaderIdExt
+        + HeaderProvider
+        + ChainSpecProvider
+        + StateProviderFactory
+        + EvmEnvProvider
+        + 'static,
+    Eth: EthApiSpec
+        + LoadState
+        + LoadPendingBlock
+        + LoadBlock
+        + Call
+        + Trace
+        + SpawnBlocking
+        + 'static,
```

Comment:
before we finilaize this pr, I'd like to explore if we could bundle all of these via a helper trait and blanket impl